### PR TITLE
fix(install): Make installing more verbose

### PR DESCRIPTION
**Pull request by nickvergessen**
## Before

Not really very verbose
```
$ occ maintenance:install --verbose
Nextcloud was successfully installed
```

## After

```
$ occ maintenance:install --verbose
2023-10-31T11:00:35+00:00 Configuring database
2023-10-31T11:00:35+00:00 Run server migrations
2023-10-31T11:00:35+00:00 Migrating schema only
2023-10-31T11:00:35+00:00 - Reading 13000Date20170705121758
2023-10-31T11:00:35+00:00 - Reading 13000Date20170718121200
2023-10-31T11:00:35+00:00 - Reading 13000Date20170814074715
2023-10-31T11:00:35+00:00 - Reading 13000Date20170919121250
2023-10-31T11:00:35+00:00 - Reading 13000Date20170926101637
2023-10-31T11:00:35+00:00 - Reading 14000Date20180129121024
2023-10-31T11:00:35+00:00 - Reading 14000Date20180404140050
2023-10-31T11:00:35+00:00 - Reading 14000Date20180516101403
2023-10-31T11:00:35+00:00 - Reading 14000Date20180518120534
2023-10-31T11:00:35+00:00 - Reading 14000Date20180522074438
2023-10-31T11:00:35+00:00 - Reading 14000Date20180626223656
2023-10-31T11:00:35+00:00 - Reading 14000Date20180710092004
2023-10-31T11:00:35+00:00 - Reading 14000Date20180712153140
2023-10-31T11:00:35+00:00 - Reading 15000Date20180926101451
2023-10-31T11:00:35+00:00 - Reading 15000Date20181015062942
2023-10-31T11:00:35+00:00 - Reading 15000Date20181029084625
2023-10-31T11:00:35+00:00 - Reading 16000Date20190207141427
2023-10-31T11:00:35+00:00 - Reading 16000Date20190212081545
2023-10-31T11:00:35+00:00 - Reading 16000Date20190427105638
2023-10-31T11:00:35+00:00 - Reading 16000Date20190428150708
2023-10-31T11:00:35+00:00 - Reading 17000Date20190514105811
2023-10-31T11:00:35+00:00 - Reading 18000Date20190920085628
2023-10-31T11:00:35+00:00 - Reading 18000Date20191014105105
2023-10-31T11:00:35+00:00 - Reading 18000Date20191204114856
2023-10-31T11:00:35+00:00 - Reading 19000Date20200211083441
2023-10-31T11:00:35+00:00 - Reading 20000Date20201109081915
2023-10-31T11:00:35+00:00 - Reading 20000Date20201109081918
2023-10-31T11:00:35+00:00 - Reading 20000Date20201109081919
2023-10-31T11:00:35+00:00 - Reading 20000Date20201111081915
2023-10-31T11:00:35+00:00 - Reading 21000Date20201120141228
2023-10-31T11:00:35+00:00 - Reading 21000Date20201202095923
2023-10-31T11:00:35+00:00 - Reading 21000Date20210119195004
2023-10-31T11:00:35+00:00 - Reading 21000Date20210309185126
2023-10-31T11:00:35+00:00 - Reading 21000Date20210309185127
2023-10-31T11:00:35+00:00 - Reading 22000Date20210216080825
2023-10-31T11:00:35+00:00 - Reading 23000Date20210721100600
2023-10-31T11:00:35+00:00 - Reading 23000Date20210906132259
2023-10-31T11:00:35+00:00 - Reading 23000Date20210930122352
2023-10-31T11:00:35+00:00 - Reading 23000Date20211203110726
2023-10-31T11:00:35+00:00 - Reading 23000Date20211213203940
2023-10-31T11:00:35+00:00 - Reading 24000Date20211210141942
2023-10-31T11:00:35+00:00 - Reading 24000Date20211213081506
2023-10-31T11:00:35+00:00 - Reading 24000Date20211213081604
2023-10-31T11:00:35+00:00 - Reading 24000Date20211222112246
2023-10-31T11:00:35+00:00 - Reading 24000Date20211230140012
2023-10-31T11:00:35+00:00 - Reading 24000Date20220131153041
2023-10-31T11:00:35+00:00 - Reading 24000Date20220202150027
2023-10-31T11:00:35+00:00 - Reading 24000Date20220404230027
2023-10-31T11:00:35+00:00 - Reading 24000Date20220425072957
2023-10-31T11:00:35+00:00 - Reading 25000Date20220515204012
2023-10-31T11:00:35+00:00 - Reading 25000Date20220602190540
2023-10-31T11:00:35+00:00 - Reading 25000Date20220905140840
2023-10-31T11:00:35+00:00 - Reading 25000Date20221007010957
2023-10-31T11:00:35+00:00 - Reading 27000Date20220613163520
2023-10-31T11:00:35+00:00 - Reading 27000Date20230309104325
2023-10-31T11:00:35+00:00 - Reading 27000Date20230309104802
2023-10-31T11:00:35+00:00 - Reading 28000Date20230616104802
2023-10-31T11:00:35+00:00 - Reading 28000Date20230728104802
2023-10-31T11:00:35+00:00 - Reading 28000Date20230803221055
2023-10-31T11:00:35+00:00 - Reading 28000Date20230906104802
2023-10-31T11:00:35+00:00 - Checking target database schema
2023-10-31T11:00:35+00:00 - Migrate database schema
2023-10-31T11:00:38+00:00 - Mark migrations as executed
2023-10-31T11:00:38+00:00 Create admin user
2023-10-31T11:00:39+00:00 Install default apps
2023-10-31T11:00:39+00:00 Installing shipped apps
2023-10-31T11:00:39+00:00 Installing files_reminders
2023-10-31T11:00:39+00:00 Migrating schema only
2023-10-31T11:00:39+00:00 - Reading 10000Date20230725162149
2023-10-31T11:00:39+00:00 - Checking target database schema
2023-10-31T11:00:39+00:00 - Migrate database schema
2023-10-31T11:00:39+00:00 - Mark migrations as executed
2023-10-31T11:00:39+00:00 Registering tasks of files_reminders
2023-10-31T11:00:39+00:00 Installing files_sharing
2023-10-31T11:00:39+00:00 Migrating schema only
2023-10-31T11:00:39+00:00 - Reading 11300Date20201120141438
2023-10-31T11:00:39+00:00 - Reading 21000Date20201223143245
2023-10-31T11:00:39+00:00 - Reading 22000Date20210216084241
2023-10-31T11:00:39+00:00 - Reading 24000Date20220208195521
2023-10-31T11:00:39+00:00 - Reading 24000Date20220404142216
2023-10-31T11:00:39+00:00 - Checking target database schema
2023-10-31T11:00:39+00:00 - Migrate database schema
2023-10-31T11:00:39+00:00 - Mark migrations as executed
2023-10-31T11:00:39+00:00 Registering tasks of files_sharing
2023-10-31T11:00:39+00:00 Installing twofactor_backupcodes
2023-10-31T11:00:39+00:00 Migrating schema only
2023-10-31T11:00:39+00:00 - Reading 1002Date20170607104347
2023-10-31T11:00:39+00:00 - Reading 1002Date20170607113030
2023-10-31T11:00:39+00:00 - Reading 1002Date20170919123342
2023-10-31T11:00:39+00:00 - Reading 1002Date20170926101419
2023-10-31T11:00:39+00:00 - Reading 1002Date20180821043638
2023-10-31T11:00:39+00:00 - Checking target database schema
2023-10-31T11:00:39+00:00 - Migrate database schema
2023-10-31T11:00:39+00:00 - Mark migrations as executed
2023-10-31T11:00:39+00:00 Registering tasks of twofactor_backupcodes
2023-10-31T11:00:39+00:00 Installing oauth2
2023-10-31T11:00:39+00:00 Migrating schema only
2023-10-31T11:00:39+00:00 - Reading 010401Date20181207190718
2023-10-31T11:00:39+00:00 - Reading 010402Date20190107124745
2023-10-31T11:00:39+00:00 - Reading 011601Date20230522143227
2023-10-31T11:00:39+00:00 - Reading 011602Date20230613160650
2023-10-31T11:00:39+00:00 - Reading 011603Date20230620111039
2023-10-31T11:00:39+00:00 - Checking target database schema
2023-10-31T11:00:39+00:00 - Migrate database schema
2023-10-31T11:00:39+00:00 - Mark migrations as executed
2023-10-31T11:00:39+00:00 Registering tasks of oauth2
2023-10-31T11:00:40+00:00 Installing user_status
2023-10-31T11:00:40+00:00 Migrating schema only
2023-10-31T11:00:40+00:00 - Reading 0001Date20200602134824
2023-10-31T11:00:40+00:00 - Reading 0002Date20200902144824
2023-10-31T11:00:40+00:00 - Reading 1000Date20201111130204
2023-10-31T11:00:40+00:00 - Reading 1003Date20210809144824
2023-10-31T11:00:40+00:00 - Reading 1008Date20230921144701
2023-10-31T11:00:40+00:00 - Checking target database schema
2023-10-31T11:00:40+00:00 - Migrate database schema
2023-10-31T11:00:40+00:00 - Mark migrations as executed
2023-10-31T11:00:40+00:00 Registering tasks of user_status
2023-10-31T11:00:40+00:00 Installing files_versions
2023-10-31T11:00:40+00:00 Migrating schema only
2023-10-31T11:00:40+00:00 - Reading 1020Date20221114144058
2023-10-31T11:00:40+00:00 - Checking target database schema
2023-10-31T11:00:40+00:00 - Migrate database schema
2023-10-31T11:00:40+00:00 - Mark migrations as executed
2023-10-31T11:00:40+00:00 Registering tasks of files_versions
2023-10-31T11:00:40+00:00 Installing sharebymail
2023-10-31T11:00:40+00:00 Migrating schema only
2023-10-31T11:00:40+00:00 Registering tasks of sharebymail
2023-10-31T11:00:40+00:00 Installing cloud_federation_api
2023-10-31T11:00:40+00:00 Migrating schema only
2023-10-31T11:00:40+00:00 Registering tasks of cloud_federation_api
2023-10-31T11:00:40+00:00 Installing theming
2023-10-31T11:00:40+00:00 Migrating schema only
2023-10-31T11:00:40+00:00 Registering tasks of theming
2023-10-31T11:00:40+00:00 Installing settings
2023-10-31T11:00:40+00:00 Migrating schema only
2023-10-31T11:00:40+00:00 Registering tasks of settings
2023-10-31T11:00:40+00:00 Installing files_trashbin
2023-10-31T11:00:40+00:00 Migrating schema only
2023-10-31T11:00:40+00:00 - Reading 1010Date20200630192639
2023-10-31T11:00:40+00:00 - Checking target database schema
2023-10-31T11:00:40+00:00 - Migrate database schema
2023-10-31T11:00:40+00:00 - Mark migrations as executed
2023-10-31T11:00:40+00:00 Registering tasks of files_trashbin
2023-10-31T11:00:40+00:00 Installing systemtags
2023-10-31T11:00:40+00:00 Migrating schema only
2023-10-31T11:00:40+00:00 Registering tasks of systemtags
2023-10-31T11:00:40+00:00 Installing contactsinteraction
2023-10-31T11:00:40+00:00 Migrating schema only
2023-10-31T11:00:40+00:00 - Reading 010000Date20200304152605
2023-10-31T11:00:40+00:00 - Checking target database schema
2023-10-31T11:00:40+00:00 - Migrate database schema
2023-10-31T11:00:40+00:00 - Mark migrations as executed
2023-10-31T11:00:40+00:00 Registering tasks of contactsinteraction
2023-10-31T11:00:40+00:00 Installing provisioning_api
2023-10-31T11:00:40+00:00 Migrating schema only
2023-10-31T11:00:40+00:00 Registering tasks of provisioning_api
2023-10-31T11:00:40+00:00 Installing dashboard
2023-10-31T11:00:40+00:00 Migrating schema only
2023-10-31T11:00:40+00:00 Registering tasks of dashboard
2023-10-31T11:00:40+00:00 Installing workflowengine
2023-10-31T11:00:40+00:00 Migrating schema only
2023-10-31T11:00:40+00:00 - Reading 2000Date20190808074233
2023-10-31T11:00:40+00:00 - Reading 2200Date20210805101925
2023-10-31T11:00:40+00:00 - Checking target database schema
2023-10-31T11:00:40+00:00 - Migrate database schema
2023-10-31T11:00:41+00:00 - Mark migrations as executed
2023-10-31T11:00:41+00:00 Registering tasks of workflowengine
2023-10-31T11:00:41+00:00 Installing comments
2023-10-31T11:00:41+00:00 Migrating schema only
2023-10-31T11:00:41+00:00 Registering tasks of comments
2023-10-31T11:00:41+00:00 Installing federation
2023-10-31T11:00:41+00:00 Migrating schema only
2023-10-31T11:00:41+00:00 - Reading 1010Date20200630191302
2023-10-31T11:00:41+00:00 - Checking target database schema
2023-10-31T11:00:41+00:00 - Migrate database schema
2023-10-31T11:00:41+00:00 - Mark migrations as executed
2023-10-31T11:00:41+00:00 Registering tasks of federation
2023-10-31T11:00:41+00:00 Installing federatedfilesharing
2023-10-31T11:00:41+00:00 Migrating schema only
2023-10-31T11:00:41+00:00 - Reading 1010Date20200630191755
2023-10-31T11:00:41+00:00 - Reading 1011Date20201120125158
2023-10-31T11:00:41+00:00 - Checking target database schema
2023-10-31T11:00:41+00:00 - Migrate database schema
2023-10-31T11:00:41+00:00 - Mark migrations as executed
2023-10-31T11:00:41+00:00 Registering tasks of federatedfilesharing
2023-10-31T11:00:41+00:00 Installing updatenotification
2023-10-31T11:00:41+00:00 Migrating schema only
2023-10-31T11:00:41+00:00 Registering tasks of updatenotification
2023-10-31T11:00:41+00:00 Installing files
2023-10-31T11:00:41+00:00 Migrating schema only
2023-10-31T11:00:41+00:00 - Reading 11301Date20191205150729
2023-10-31T11:00:41+00:00 - Reading 12101Date20221011153334
2023-10-31T11:00:41+00:00 - Checking target database schema
2023-10-31T11:00:41+00:00 - Migrate database schema
2023-10-31T11:00:41+00:00 - Mark migrations as executed
2023-10-31T11:00:41+00:00 Registering tasks of files
2023-10-31T11:00:41+00:00 Installing weather_status
2023-10-31T11:00:41+00:00 Migrating schema only
2023-10-31T11:00:41+00:00 Registering tasks of weather_status
2023-10-31T11:00:41+00:00 Installing dav
2023-10-31T11:00:41+00:00 Migrating schema only
2023-10-31T11:00:41+00:00 - Reading 1004Date20170825134824
2023-10-31T11:00:41+00:00 - Reading 1004Date20170919104507
2023-10-31T11:00:41+00:00 - Reading 1004Date20170924124212
2023-10-31T11:00:41+00:00 - Reading 1004Date20170926103422
2023-10-31T11:00:41+00:00 - Reading 1005Date20180413093149
2023-10-31T11:00:41+00:00 - Reading 1005Date20180530124431
2023-10-31T11:00:41+00:00 - Reading 1006Date20180619154313
2023-10-31T11:00:41+00:00 - Reading 1006Date20180628111625
2023-10-31T11:00:41+00:00 - Reading 1008Date20181030113700
2023-10-31T11:00:41+00:00 - Reading 1008Date20181105104826
2023-10-31T11:00:41+00:00 - Reading 1008Date20181105104833
2023-10-31T11:00:41+00:00 - Reading 1008Date20181105110300
2023-10-31T11:00:41+00:00 - Reading 1008Date20181105112049
2023-10-31T11:00:41+00:00 - Reading 1008Date20181114084440
2023-10-31T11:00:41+00:00 - Reading 1011Date20190725113607
2023-10-31T11:00:41+00:00 - Reading 1011Date20190806104428
2023-10-31T11:00:41+00:00 - Reading 1012Date20190808122342
2023-10-31T11:00:41+00:00 - Reading 1016Date20201109085907
2023-10-31T11:00:41+00:00 - Reading 1017Date20210216083742
2023-10-31T11:00:41+00:00 - Reading 1018Date20210312100735
2023-10-31T11:00:41+00:00 - Reading 1024Date20211221144219
2023-10-31T11:00:41+00:00 - Reading 1027Date20230504122946
2023-10-31T11:00:41+00:00 - Reading 1029Date20231004091403
2023-10-31T11:00:41+00:00 - Checking target database schema
2023-10-31T11:00:41+00:00 - Migrate database schema
2023-10-31T11:00:43+00:00 - Mark migrations as executed
2023-10-31T11:00:43+00:00 Registering tasks of dav
2023-10-31T11:00:43+00:00 Installing lookup_server_connector
2023-10-31T11:00:43+00:00 Migrating schema only
2023-10-31T11:00:43+00:00 Registering tasks of lookup_server_connector
2023-10-31T11:00:43+00:00 Installing notifications
2023-10-31T11:00:43+00:00 Migrating schema only
2023-10-31T11:00:43+00:00 - Reading 2004Date20190107135757
2023-10-31T11:00:43+00:00 - Reading 2010Date20210218082811
2023-10-31T11:00:43+00:00 - Reading 2010Date20210218082855
2023-10-31T11:00:43+00:00 - Reading 2011Date20210930134607
2023-10-31T11:00:43+00:00 - Reading 2011Date20220826074907
2023-10-31T11:00:43+00:00 - Checking target database schema
2023-10-31T11:00:43+00:00 - Migrate database schema
2023-10-31T11:00:43+00:00 - Mark migrations as executed
2023-10-31T11:00:43+00:00 Registering tasks of notifications
2023-10-31T11:00:43+00:00 Installing support
2023-10-31T11:00:43+00:00 Migrating schema only
2023-10-31T11:00:43+00:00 Registering tasks of support
2023-10-31T11:00:43+00:00 Installing privacy
2023-10-31T11:00:43+00:00 Migrating schema only
2023-10-31T11:00:43+00:00 - Reading 100Date20190217131943
2023-10-31T11:00:43+00:00 - Checking target database schema
2023-10-31T11:00:43+00:00 - Migrate database schema
2023-10-31T11:00:43+00:00 - Mark migrations as executed
2023-10-31T11:00:43+00:00 Registering tasks of privacy
2023-10-31T11:00:44+00:00 Installing related_resources
2023-10-31T11:00:44+00:00 Migrating schema only
2023-10-31T11:00:44+00:00 Registering tasks of related_resources
2023-10-31T11:00:44+00:00 Installing password_policy
2023-10-31T11:00:44+00:00 Migrating schema only
2023-10-31T11:00:44+00:00 Registering tasks of password_policy
2023-10-31T11:00:44+00:00 Installing nextcloud_announcements
2023-10-31T11:00:44+00:00 Migrating schema only
2023-10-31T11:00:44+00:00 Registering tasks of nextcloud_announcements
2023-10-31T11:00:44+00:00 Installing files_pdfviewer
2023-10-31T11:00:44+00:00 Migrating schema only
2023-10-31T11:00:44+00:00 Registering tasks of files_pdfviewer
2023-10-31T11:00:44+00:00 Installing text
2023-10-31T11:00:44+00:00 Migrating schema only
2023-10-31T11:00:44+00:00 - Reading 010000Date20190617184535
2023-10-31T11:00:44+00:00 - Reading 030001Date20200402075029
2023-10-31T11:00:44+00:00 - Reading 030201Date20201116110353
2023-10-31T11:00:44+00:00 - Reading 030201Date20201116123153
2023-10-31T11:00:44+00:00 - Reading 030501Date20220202101853
2023-10-31T11:00:44+00:00 - Reading 030701Date20230207131313
2023-10-31T11:00:44+00:00 - Checking target database schema
2023-10-31T11:00:44+00:00 - Migrate database schema
2023-10-31T11:00:44+00:00 - Mark migrations as executed
2023-10-31T11:00:44+00:00 Registering tasks of text
2023-10-31T11:00:44+00:00 Installing photos
2023-10-31T11:00:44+00:00 Migrating schema only
2023-10-31T11:00:44+00:00 - Reading 20000Date20220727125801
2023-10-31T11:00:44+00:00 - Reading 20001Date20220830131446
2023-10-31T11:00:44+00:00 - Reading 20003Date20221102170153
2023-10-31T11:00:44+00:00 - Reading 20003Date20221103094628
2023-10-31T11:00:44+00:00 - Checking target database schema
2023-10-31T11:00:44+00:00 - Migrate database schema
2023-10-31T11:00:44+00:00 - Mark migrations as executed
2023-10-31T11:00:44+00:00 Registering tasks of photos
2023-10-31T11:00:44+00:00 Installing serverinfo
2023-10-31T11:00:44+00:00 Migrating schema only
2023-10-31T11:00:44+00:00 Registering tasks of serverinfo
2023-10-31T11:00:44+00:00 Installing recommendations
2023-10-31T11:00:44+00:00 Migrating schema only
2023-10-31T11:00:44+00:00 Registering tasks of recommendations
2023-10-31T11:00:44+00:00 Installing firstrunwizard
2023-10-31T11:00:44+00:00 Migrating schema only
2023-10-31T11:00:44+00:00 Registering tasks of firstrunwizard
2023-10-31T11:00:44+00:00 Installing activity
2023-10-31T11:00:44+00:00 Migrating schema only
2023-10-31T11:00:44+00:00 - Reading 2006Date20170808154933
2023-10-31T11:00:44+00:00 - Reading 2006Date20170808155040
2023-10-31T11:00:44+00:00 - Reading 2006Date20170919095939
2023-10-31T11:00:44+00:00 - Reading 2007Date20181107114613
2023-10-31T11:00:44+00:00 - Reading 2008Date20181011095117
2023-10-31T11:00:44+00:00 - Reading 2010Date20190416112817
2023-10-31T11:00:44+00:00 - Reading 2011Date20201006132544
2023-10-31T11:00:44+00:00 - Reading 2011Date20201006132545
2023-10-31T11:00:44+00:00 - Reading 2011Date20201006132546
2023-10-31T11:00:44+00:00 - Reading 2011Date20201006132547
2023-10-31T11:00:44+00:00 - Reading 2011Date20201207091915
2023-10-31T11:00:44+00:00 - Checking target database schema
2023-10-31T11:00:44+00:00 - Migrate database schema
2023-10-31T11:00:45+00:00 - Mark migrations as executed
2023-10-31T11:00:45+00:00 Registering tasks of activity
2023-10-31T11:00:45+00:00 Installing files_rightclick
2023-10-31T11:00:45+00:00 Migrating schema only
2023-10-31T11:00:45+00:00 Registering tasks of files_rightclick
2023-10-31T11:00:45+00:00 Installing survey_client
2023-10-31T11:00:45+00:00 Migrating schema only
2023-10-31T11:00:45+00:00 Registering tasks of survey_client
2023-10-31T11:00:45+00:00 Installing logreader
2023-10-31T11:00:45+00:00 Migrating schema only
2023-10-31T11:00:45+00:00 Registering tasks of logreader
2023-10-31T11:00:45+00:00 Installing viewer
2023-10-31T11:00:45+00:00 Migrating schema only
2023-10-31T11:00:45+00:00 Registering tasks of viewer
2023-10-31T11:00:45+00:00 Installing circles
2023-10-31T11:00:45+00:00 Migrating schema only
2023-10-31T11:00:45+00:00 - Reading 0022Date20220526111723
2023-10-31T11:00:45+00:00 - Reading 0022Date20220526113601
2023-10-31T11:00:45+00:00 - Reading 0022Date20220703115023
2023-10-31T11:00:45+00:00 - Reading 0023Date20211216113101
2023-10-31T11:00:45+00:00 - Reading 0024Date20220203123901
2023-10-31T11:00:45+00:00 - Reading 0024Date20220203123902
2023-10-31T11:00:45+00:00 - Reading 0024Date20220317190331
2023-10-31T11:00:45+00:00 - Reading 0028Date20230705222601
2023-10-31T11:00:45+00:00 - Checking target database schema
2023-10-31T11:00:45+00:00 - Migrate database schema
2023-10-31T11:00:46+00:00 - Mark migrations as executed
2023-10-31T11:00:46+00:00 Registering tasks of circles
2023-10-31T11:00:46+00:00 Setup data directory
2023-10-31T11:00:46+00:00 Install background jobs
2023-10-31T11:00:52+00:00 Nextcloud was successfully installed
```

## Checklist

- Code is [properly formatted](https://docs.nextcloud.com/server/latest/developer_manual/digging_deeper/continuous_integration.html#linting)
- [Sign-off message](https://github.com/src-d/guide/blob/master/developer-community/fix-DCO.md) is added to all commits
- [x] Tests ([unit](https://docs.nextcloud.com/server/latest/developer_manual/app_development/tutorial.html#unit-tests), [integration](https://docs.nextcloud.com/server/latest/developer_manual/app_development/tutorial.html#integration-tests), api and/or acceptance) are included
- [x] Screenshots before/after for front-end changes
- [x] Documentation ([manuals](https://github.com/nextcloud/documentation/) or wiki) has been updated or is not required
- [x] [Backports requested](https://github.com/nextcloud/backportbot/#usage) where applicable (ex: critical bugfixes)

### CODE REVIEW

File: lib/public/Migration/IOutput.php [hunk 1]
```diff
@@ -29,6 +29,13 @@
  * @since 9.1.0
  */
 interface IOutput {
+	/**
+	 * @param string $message
+	 * @return void
+	 * @since 28.0.0
+	 */
+	public function debug($message);
```
Comment (ChristophWurst) [inline lib/public/Migration/IOutput.php:37]:
```suggestion
	public function debug(string $message): void;
```

**Review by ChristophWurst — APPROVED** [review]
Inconsistent with the other methods but let's add the typing for new methods?

**Review by kesselb — APPROVED** [review]
Good stuff :+1:

**Comment by kesselb** [comment]
I assume we could simplify the code a bit by using something like

```
$output->writeln(
            'Will only be printed in verbose mode or higher',
            OutputInterface::VERBOSITY_VERBOSE
        );
```

and always passing an IOutput instance, however the current approach works as well.

**Comment by nickvergessen** [comment]
> and always passing an IOutput instance, however the current approach works as well.

Well then we need to also inject it always from the SetupController.

**Comment by kesselb** [comment]
> Well then we need to also inject it always from the SetupController.

 I see. My suggestion was to simplify the code a bit and don't have different paths. Indeed, for the web-based setup there is not much use of it as we don't have a verbose mode. I assume we can use a NullOutput there.

Nevertheless, the current approach is also 100% fine and a good improvement already :+1:

**Pull request merged**